sphinx/tutorial/cylc/furthertopics/suicide-triggers.rst

```diff
@@ -0,0 +1,654 @@
+   "bake_cake.3" -> "eat_cake.3"
+
+The reason the suite stalls is that, by default, cylc will run a maximum of
+three cycles concurrently. As each cycle has at least one task which hasn't
+either succeeded or failed cylc cannot move onto the next cycle.
+
+.. tip::
+   
+   For more information search ``max active cycle points`` in the
+   `cylc user guide`_.
+
+You will also notice that some of the tasks (e.g. ``eat_cake`` in cycle ``2``
+in the above example) are drawn in a faded gray. This is because these tasks
+have not yet been run in earlier cycles and as such cannot run.
+
+.. TODO - Spawn On Demand!
+
+
+Removing Tasks From The Graph
+-----------------------------
+
+In order to get around these problems and prevent the suite from stalling we
+must remove the tasks that are no longer needed. We do this using suicide
+triggers.
+
+A suicide trigger is written like a normal dependency but with an exclamation
+mark in-front of the task on the right-hand-side of the dependency meaning
+*"remove the following task from the graph at the current cycle point."*
+
+For example the following :term:`graph string` would remove the task ``bar``
+from the graph if the task ``foo`` were to succeed.
+
+.. code-block:: cylc-graph
+
+   foo => ! bar
+
+There are three cases where we would need to remove a task in the cake-making
+example:
+
+#. If the ``bake_cake`` task succeeds we don't need the ``eat_cake`` task so
+   should remove it.
+
+   .. code-block:: cylc-graph
+
+      bake_cake => ! eat_cake
+
+#. If the ``bake_cake`` task fails we don't need the ``sell_cake`` task so
+   should remove it.
+
+   .. code-block:: cylc-graph
+
+      bake_cake:fail => ! sell_cake
+
+#. The ``bake_cake`` task needs to be removed else the suite will stall, we can
+   do this after the ``eat_cake`` task has succeeded.
+
+   .. code-block:: cylc-graph
+
+      eat_cake => ! bake_cake
+
+Add the following three lines to the suite's graph:
+
+.. code-block:: cylc-graph
+
+   bake_cake => ! eat_cake
+   bake_cake:fail => ! sell_cake
+   eat_cake => ! bake_cake
+
+We can view suicide triggers in ``cylc graph`` by un-selecting the
+:guilabel:`Ignore Suicide Triggers` button in the toolbar. Suicide triggers
+will then appear as dashed lines with circular endings. You should see
+something like this:
+
+.. digraph:: Example
+   :align: center
+
+   bgcolor = none
+
+   make_cake_mixture -> bake_cake
+   bake_cake -> sell_cake [style="dashed" arrowhead="dot"]
+   bake_cake -> eat_cake [style="dashed" arrowhead="dot"]
+   eat_cake -> bake_cake [style="dashed" arrowhead="dot"]
+
+
+Downstream Dependencies
+-----------------------
+
+If we wanted to make the cycles run in order we might write an
+:term:`inter-cycle dependency` like this:
+
+.. code-block:: cylc-graph
+
+   sell_cake[-P1] => make_cake_mixture
+
+In order to handle the event that the ``sell_cake`` task has been removed from
+the graph by a suicide trigger we can write our dependency with an or ``|`` like
```

Comment:
I would put ``|`` in parentheses and/or put '..."or" symbol ``|``...' instead, as I had to read this a few times to understand what was meant.